pkg/client/options.go

```diff
@@ -150,8 +150,12 @@ func (o *CreateOptions) AsCreateOptions() *metav1.CreateOptions {
 		o.Raw = &metav1.CreateOptions{}
 	}
 
```

Comment:
Should we document that override behavior somewhere in the godoc of `AsCreateOptions`? (same for others of course)

I know we have `This may mutate the Raw field.` but it's very fuzzy